src/api/dirmonitor/kqueue.c

```diff
@@ -23,7 +24,9 @@ void deinit_dirmonitor(struct dirmonitor_internal* monitor) {
 
 
 int get_changes_dirmonitor(struct dirmonitor_internal* monitor, char* buffer, int buffer_size) {
-  int nev = kevent(monitor->fd, NULL, 0, (struct kevent*)buffer, buffer_size / sizeof(kevent), NULL);
+  struct timespec ts = { 0, 10 * 1000000 }; // 10 ms
+
+  int nev = kevent(monitor->fd, NULL, 0, (struct kevent*)buffer, buffer_size / sizeof(kevent), &ts);
```

Comment:
the thread does nothing except if it should check and then checks for a small amount of time.

This can be lowered down to 1ms but I feel like 10ms is fast enough while giving enough room for kevent.